import/controls/permissionfilterproxymodel.cpp

```diff
@@ -0,0 +1,68 @@
+/****************************************************************************
+**
+** Copyright (C) 2020 Open Mobile Platform LLC
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "permissionfilterproxymodel.h"
+#include "permissionmodel.h"
+#include "permissionmanager.h"
+
+PermissionFilterProxyModel::PermissionFilterProxyModel(QObject *parent)
+    : QSortFilterProxyModel(parent)
+{
+
+}
+
+void PermissionFilterProxyModel::remove(int currentIndex)
+{
+    QModelIndex proxyIndex = index(currentIndex, 0);
+    QModelIndex sourceIndex = mapToSource(proxyIndex);
+
+    PermissionModel *permissionModel = dynamic_cast<PermissionModel *>(sourceModel());
+    if (!permissionModel)
+        return;
+
+    permissionModel->remove(sourceIndex.row());
+}
+
+void PermissionFilterProxyModel::setCapability(int currentIndex, int capability)
+{
+    QModelIndex proxyIndex = index(currentIndex, 0);
+    QModelIndex sourceIndex = mapToSource(proxyIndex);
+
+    PermissionModel *permissionModel = dynamic_cast<PermissionModel *>(sourceModel());
```

Comment:
```suggestion
    PermissionModel *permissionModel = qobject_cast<PermissionModel *>(sourceModel());
```
RTTI should be avoided.